Update, including new deploy-local.py changes (#239)
bobanetwork/boba: .gitbook/contents/hc/3_write-smart-contract.md

1. Intentionally burn (or "waste") ETH gas to simulate a more true-to-life Hybrid Compute request.
2. Increment a counter variable each time the contract is called based on certain inputs and errors.

> Our contract is an extension of an example provided by the upstream Account Abstraction framework on
> which we based our Hybrid Compute extensions: [link](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/test/TestCounter.sol)

Reviewer: ```suggestion
Our contract is an extension of an [example provided](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/test/TestCounter.sol) by the upstream Account Abstraction framework on
which we based our Hybrid Compute extensions.
```

You can find the needed `HybridAccount` contract along with its dependencies in our [repository](https://github.com/bobanetwork/account-abstraction-hc/contracts/samples/HybridAccount.sol). Additionally, the contract is pulled in as a submodule when you checkout our [`rundler-hc` repository](https://github.com/bobanetwork/rundler-hc/crates/types/contracts/lib/account-abstraction/).

> The code blocks below are edited to show the lines being discussed. It is recommended to open another

Reviewer: ```suggestion
The code blocks below are edited to show the lines being discussed. We recommend opening another
```
